components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientsFetchResourceImplTest.java

```diff
@@ -0,0 +1,355 @@
+        this.component = (DefaultPatientsFetchResourceImpl) this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        this.repository = this.mocker.getInstance(PatientRepository.class, "secure");
+
+        when(this.patient1.getId()).thenReturn(ID_1);
+        when(this.patient1.getDocument()).thenReturn(this.patientReference1);
+        when(this.patient2.getId()).thenReturn(ID_2);
+        when(this.patient2.getDocument()).thenReturn(this.patientReference2);
+        when(this.patient3.getId()).thenReturn(ID_3);
+        when(this.patient3.getDocument()).thenReturn(this.patientReference3);
+
+        final Autolinker autolinker = this.mocker.getInstance(Autolinker.class);
+        when(autolinker.forSecondaryResource(any(Class.class), any(UriInfo.class))).thenReturn(autolinker);
+        when(autolinker.withExtraParameters(any(String.class), any(String.class))).thenReturn(autolinker);
+        doReturn(this.uriList).when(autolinker).build();
+    }
+
+    @Test
+    public void getPatientsWithEmptyEidAndId() throws ComponentLookupException
+    {
+        final Request request = mock(Request.class);
+        doReturn(request).when(this.container).getRequest();
+        doReturn(Collections.emptyList()).when(request).getProperties("eid");
+        doReturn(Collections.emptyList()).when(request).getProperties("id");
+
+        final Response response = this.component.fetchPatients();
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+        assertEquals("[]", response.getEntity());
+    }
+
+    @Test
+    public void getPatientsPerformsCorrectlyOnePatientRecordByEidNoneById()
+        throws ComponentLookupException, QueryException, IOException
+    {
+        final List<Object> eidList = ImmutableList.<Object>of(EID1);
+
+        final Request request = mock(Request.class);
+        doReturn(request).when(this.container).getRequest();
+        doReturn(eidList).when(request).getProperties("eid");
+        doReturn(Collections.emptyList()).when(request).getProperties("id");
+
+        final Query query = mock(DefaultQuery.class);
+        doReturn(this.patient1JSON).when(this.patient1).toJSON();
+        doReturn(query).when(this.qm).createQuery(Matchers.anyString(), Matchers.anyString());
+        doReturn(ImmutableList.<Object>of(ID_1)).when(query).execute();
+
+        doReturn(this.patient1).when(this.repository).get(ID_1);
+        doReturn(ID_1).when(this.patient1).getId();
+
+        final Response response = this.component.fetchPatients();
+
+        assertEquals(Response.Status.OK.getStatusCode(), response.getStatus());
+
+        final String expectedStr = new JSONArray().put(new JSONObject().put("id", ID_1).put("links", this.uriList))
+            .toString();
+        @SuppressWarnings("unchecked")
+        final List<String> expected = (List<String>) (OBJECT_MAPPER.readValue(expectedStr, List.class));
```

Comment:
I think you can do `assertTrue(expectedJson.similar(actualJson))` to simplify this kind of check.